src/ticks/exp_engine.c

```diff
@@ -0,0 +1,517 @@
+            break;
+        }
+    }
+}
+
+void expression_math_div(struct expression_result_t* a, struct expression_result_t* b, struct expression_result_t* result) {
+    if (!(are_exp_same_type(a, &a->type, b, &b->type))) {
+        struct expression_result_t local_3;
+        convert_expression(b, &local_3, a->type.type_, is_expression_signed(b));
+        expression_math_div(a, &local_3, result);
+        return;
+    }
+
+    switch (a->type.type_) {
+        case TYPE_FLOAT: {
+            result->as_float = a->as_float / b->as_float;
+            return;
+        }
+        case TYPE_CHAR:
+        case TYPE_INT:
+        case TYPE_LONG: {
+            if (is_expression_signed(a)) {
+                result->as_int = a->as_int / b->as_int;
+            } else {
+                result->as_uint = (uint32_t)((uint32_t)a->as_int / (uint32_t)b->as_int);
+            }
+            return;
+        }
+        default: {
+            result->flags |= EXPRESSION_ERROR;
+            char tp[128];
+            expression_result_type_to_string(&a->type, is_expression_signed(a), tp);
+            sprintf(result->as_error, "Cannot perform math '+' on type %s", tp);
+            break;
+        }
+    }
+}
+
+uint8_t is_primitive_integer_type(struct expression_result_t* exp) {
+    switch (exp->type.type_) {
+        case TYPE_CHAR:
+        case TYPE_INT:
+        case TYPE_LONG: {
+            return 1;
+        }
+        default: {
+            return 0;
+        }
+    }
+}
+
+void expression_result_value_to_string(struct expression_result_t* result, char* buffer) {
+    switch (result->type.type_) {
+        case TYPE_CHAR:
+        case TYPE_INT:
+        case TYPE_LONG: {
+            if (is_expression_signed(result)) {
+                sprintf(buffer, "%i", result->as_int);
+            } else {
+                sprintf(buffer, "%u", result->as_uint);
+            }
+            break;
+        }
+        case TYPE_FLOAT: {
+            sprintf(buffer, "%f", result->as_float);
+            break;
+        }
+        case TYPE_STRUCTURE: {
+            sprintf(buffer, "{%#04x}", result->as_pointer.ptr);
+            break;
+        }
+        case TYPE_GENERIC_POINTER:
+        case TYPE_CODE_POINTER: {
+            if (result->type.next == NULL) {
+                sprintf(buffer, "%#04x", result->as_pointer.ptr);
+                break;
+            }
+            switch (result->type.next->type_) {
+                case TYPE_INT:
+                case TYPE_LONG:
+                {
+                    struct expression_result_t local = {};
+                    expression_dereference_pointer(result, &local);
+                    char buff[128];
+                    expression_result_value_to_string(&local, buff);
+                    sprintf(buffer, "%#04x(%s)", result->as_pointer.ptr, buff);
+                    break;
+                }
+                case TYPE_CHAR: {
```

Comment:
This one needs to isprint() the character or we will end up trashing the terminal